### PR TITLE
CodeActAgent: Fix iteration reminder

### PR DESCRIPTION
**Pull request by li-boxuan**
This is a follow-up of #1760. This PR includes three changes:
1) Iteration reminder should start with MAX_ITERATIONS from config rather than default value 100
2) In the first prompt, we should tell the LLM it has `MAX_ITERATIONS - 1` turns left, rather than `MAX_ITERATIONS - 2`
3) Remove legacy ITERATION_REMINDER config

I verified the integration tests would fail without the changes to `tests/integration/mock/CodeActAgent/test_edits/prompt_*.log` files.

### CODE REVIEW

File: tests/integration/regenerate.sh [hunk 1]
```diff
@@ -109,9 +102,6 @@ for ((i = 0; i < num_of_tests; i++)); do
         WORKSPACE_BASE=$WORKSPACE_BASE \
         DEBUG=true \
         WORKSPACE_MOUNT_PATH=$WORKSPACE_MOUNT_PATH AGENT=$agent \
-        REMIND_ITERATIONS=$remind_iterations \
-        WORKSPACE_MOUNT_PATH=$WORKSPACE_MOUNT_PATH \
-        AGENT=$agent \
```
Comment (li-boxuan) [inline tests/integration/regenerate.sh:114]: this duplicated env variable definition must be from merge conflict resolution.

**Comment by li-boxuan** [comment]
I verified the integration tests would fail without the changes to `tests/integration/mock/CodeActAgent/test_edits/prompt_*.log` files.

**Comment by li-boxuan** [comment]
It would be interesting to see if this bug would actually make a difference to benchmarks. If we fool the agent that it has more/fewer available turns, would it perform differently?

**Review by xingyaoww — APPROVED** [review]
LGTM! That's quick -- i also spotted this issue but haven't got time for fixes :)

**Pull request merged**